yjit/src/codegen.rs

```diff
@@ -9922,6 +9922,34 @@ fn gen_objtostring(
         );
 
         // No work needed. The string value is already on the top of the stack.
+        Some(KeepCompiling)
+    } else if unsafe { RB_TYPE_P(comptime_recv, RUBY_T_SYMBOL) } && assume_method_basic_definition(jit, asm, comptime_recv.class_of(), ID!(to_s)) {
+        jit_guard_known_klass(
+            jit,
+            asm,
+            comptime_recv.class_of(),
+            recv,
+            recv.into(),
+            comptime_recv,
+            SEND_MAX_DEPTH,
+            Counter::objtostring_not_string,
+        );
+
+        extern "C" {
+            fn rb_sym2str(sym: VALUE) -> VALUE;
+        }
+
+        // Same optimization done in the interpreter: rb_sym_to_s() allocates a mutable string, but since we are only
+        // going to use this string for interpolation, it's fine to use the
+        // frozen string.
+        let sym = recv;
+        let str = asm.ccall(rb_sym2str as *const u8, vec![sym]);
+        asm.stack_pop(1);
```

Comment:
@XrXr would we need `jit_prepare_call_with_gc` here? I'm assuming that `rb_sym2str` will dynamically allocate new strings into a table? Or are those strings pre-allocated when each symbol is created?

Reply:
This function is alloc-free (I just rewrote the docs for it the [other day](https://github.com/ruby/ruby/commit/88764dde78ee3475cd61447e45945a6569e8d639)!) Worth mentioning this fact in comment, though.

Reply:
I don't think we need it. I also added a debug assertion last week that _should_ catch cases where we missed `jit_prepare_call_with_gc` in more cases https://github.com/ruby/ruby/pull/12204